pkg/proxy/policy/policy_test.go

```diff
@@ -103,6 +103,21 @@ func TestApplyOnURLMatchRule(t *testing.T) {
 			url:      "https://host.com/path/with/prefix",
 			wantResp: http.StatusOK,
 		},
+		{
+			name: "host policy rule matches url host suffix",
+			policy: Policy{
+				AnyOf: []Rule{
+					URLMatchRule{
+						Host:      "host.com",
+						HostMatch: SuffixMatch,
+						Path:      "/path",
+						PathMatch: PrefixMatch,
+					},
+				},
+			},
+			url:      "https://sub.host.com/path/with/prefix",
```

Comment:
I think my comment may have gotten eaten first time, but would you mind adding another test case for `nothost.com`?

Reply:
Done